### ISSUE TITLE
Blockly evaluates times entered without leading zero on the hour as a different time

### ISSUE THREAD

**Issue by rwaaren** [domoticz/domoticz#3354]
Blockly evaluates time = 0:25 as true on 0:05 and 6:25 on 6:05.
When entered as 00:25 / 06:25 the evaluation to true is at the expected time.
Tried to find where this happens in source but have not found it yet. Please advise where conversion from string to minutes after midnight happens.  

See [this forum topic](https://www.domoticz.com/forum/viewtopic.php?f=62&t=28470] ) for detailed description and Blockly screens

**Comment by rwaaren** [comment]
Think the evaluation takes place in 
 www/app/events/blockly_blocks_domoticz.js (function Blockly.Blocks['logic_timevalue.TimeValidator'] )

Tried something but `if (text.length < 5 ) { text = str.concat('0', text ) } ;
`
does not do anything. Hopefully someone with js knowledge can do this with more succes

**Comment by gizmocuz** [comment]
I think this has to be solved in EventSystem.cpp ?

If i create the following blockly:

![image](https://user-images.githubusercontent.com/2265299/62006526-c9508a00-b141-11e9-855f-bf07f1aa18b5.png)

the following code is generated

![image](https://user-images.githubusercontent.com/2265299/62006532-d79ea600-b141-11e9-8f26-3a021758a0cb.png)

That looks good ?

**Comment by rwaaren** [comment]
The issue is only there if you leave out the leading zero with times < 10:00

57|32|timeofday  ==  365|commandArray["WriteToLogText"]="6:15"|1
58|33|timeofday  ==  375|commandArray["WriteToLogText"]="06:15"|1


`<xml xmlns="http://www.w3.org/1999/xhtml"><block type="domoticzcontrols_if" id="m:*IUH{cuS2BZU*~B8Bz" x="6" y="5"><value name="IF0"><block type="logic_timeofday" id="g[4pEa5e8*8yDa-w^wXY"><field name="OP">EQ</field><value name="Time"><block type="logic_timevalue" id="w|l4@%Zk2xq;NZ-hRTCl"><field name="TEXT">6:15</field></block></value></block></value><statement name="DO0"><block type="writetolog" id="Mg6=HnX`e@{cF8Fv]^pm"><value name="writeToLog"><block type="text" id="@}hoyLuhmM;*e@sl/)Xl"><field name="TEXT">6:15</field></block></value></block></statement></block></xml>`

`<xml xmlns="http://www.w3.org/1999/xhtml"><block type="domoticzcontrols_if" id="m:*IUH{cuS2BZU*~B8Bz" x="6" y="5"><value name="IF0"><block type="logic_timeofday" id="g[4pEa5e8*8yDa-w^wXY"><field name="OP">EQ</field><value name="Time"><block type="logic_timevalue" id="w|l4@%Zk2xq;NZ-hRTCl"><field name="TEXT">06:15</field></block></value></block></value><statement name="DO0"><block type="writetolog" id="Mg6=HnX`e@{cF8Fv]^pm"><value name="writeToLog"><block type="text" id="@}hoyLuhmM;*e@sl/)Xl"><field name="TEXT">06:15</field></block></value></block></statement></block></xml>`

**Comment by gizmocuz** [comment]
Okey solved by changing the code like this:

```
            if (timeBlock.attr('type') == 'logic_timevalue') {
                var valueA = $(timeBlock).children('field[name=\'TEXT\']')[0];
                var totalminutes = 9999;
                var res = $(valueA).text().split(":");
                if (res.length == 2) {
					var hours = parseInt(res[0]);
					var minutes = parseInt(res[1]);
					totalminutes = (hours * 60) + minutes;
                }
                compareString = 'timeofday ' + locOperand + ' ' + totalminutes;
            }

```

Will be in the upcoming beta, events have to be resaved

**Comment by rwaaren** [comment]
Solved in V4.11060. Thx !
